File: API/Helpers/UserPrams.cs
namespace API.Helpers
{
    public class UserPrams
    {
        public int PageNumber { get; set; }=1;
        private const int MaxPageSize = 50;
        private int _PageSize = 10;
        public int PageSize
        {
            get => _PageSize;
            set => _PageSize = (value > MaxPageSize) ? MaxPageSize : value;
        }
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; }=18;
        public int MaxAge { get; set; }=150;
    }
}

File: client/src/app/_models/userprams.ts
import { User } from "./user";

export class UserPrams{
    gender:string;
    pageNumber=1;
    pageSize=5;
    minAge=18;
    maxAge=99;
    orderBy='lastActive';

    constructor(user:User) {
        this.gender=user.gender;
    }
}